### PR TITLE
AV-223238: Auto annotate http route svc backend with NPL enabled

### PR DESCRIPTION
**Pull request by ashishnayak96**

### CODE REVIEW

**Comment by ashishnayak96** [comment]
build ako

File: ako-gateway-api/nodes/dequeue_ingestion.go [hunk 1]
```diff
@@ -57,6 +57,19 @@ func DequeueIngestion(key string, fullsync bool) {
...
+		_, err := utils.GetInformers().ServiceInformer.Lister().Services(namespace).Get(name)
+		if err != nil {
+			if k8serrors.IsNotFound(err) {
+				objects.SharedClusterIpLister().Delete(namespace + "/" + name)
+			} else {
+				utils.AviLog.Errorf("key: %s, msg: got error while getting service object", key)
+				return
+			}
+		}
+		objects.SharedClusterIpLister().Save(namespace+"/"+name, name)
```
Comment (aaha97) [inline ako-gateway-api/nodes/dequeue_ingestion.go:70]: where are we using this SharedClusterIpLister?

File: internal/status/svc_annotation.go [hunk 2]
```diff
@@ -28,7 +28,7 @@ import (
 
 func CheckNPLSvcAnnotation(key, namespace, name string) bool {
 	service, err := utils.GetInformers().ServiceInformer.Lister().Services(namespace).Get(name)
-	if err != nil || service.Spec.Type == corev1.ServiceTypeNodePort {
```
Comment (aaha97) [inline internal/status/svc_annotation.go:31]: why is this condition removed? we don't annotate ServiceTypeNodePort

**Comment by ashishnayak96** [comment]
build ako

File: tests/gatewayapitests/npltests/gateway_npl_test.go [hunk 3]
```diff
@@ -450,3 +451,176 @@ func TestDelSvc(t *testing.T) {
...
+	// check annotation present
+
+	// create route 2
+	// delete route 1
+	// check annotation present
+
+	// delete route 2
+	// check annotation not present
+
+	gatewayName := "gateway-npl-06"
```
[the remark this answers is not in the document]
Comment (ashishnayak96) [reply tests/gatewayapitests/npltests/gateway_npl_test.go]: done

[the remark this answers is not in the document]
Comment (ashishnayak96) [reply tests/gatewayapitests/npltests/gateway_npl_test.go]: done

File: tests/gatewayapitests/npltests/gateway_npl_test.go [hunk 4]
```diff
@@ -450,3 +451,176 @@ func TestDelSvc(t *testing.T) {
...
+
+		if !found {
+			return 0
+		}
+		nodes := aviModel.(*avinodes.AviObjectGraph).GetAviEvhVS()
+		return len(nodes[0].EvhNodes)
+	}, 25*time.Second).Should(gomega.Equal(1))
+
+	tests.TeardownHTTPRoute(t, httpRouteName, DEFAULT_NAMESPACE)
+	g.Eventually(func() bool {
```
[the remark this answers is not in the document]
Comment (ashishnayak96) [reply tests/gatewayapitests/npltests/gateway_npl_test.go:555]: added

File: tests/gatewayapitests/npltests/gateway_npl_test.go [hunk 5]
```diff
@@ -450,3 +451,176 @@ func TestDelSvc(t *testing.T) {
...
+	}, 5*time.Second).Should(gomega.Equal(true))
+
+	integrationtest.DelSVC(t, "default", "avisvc")
+
+	tests.TeardownGateway(t, gatewayName, DEFAULT_NAMESPACE)
+	tests.TeardownGatewayClass(t, gatewayClassName)
+}
+
+func TestSvcUpdateAutoAnnotate(t *testing.T) {
+	gatewayName := "gateway-npl-06"
```
[the remark this answers is not in the document]
Comment (ashishnayak96) [reply tests/gatewayapitests/npltests/gateway_npl_test.go]: fixed

**Review by pkoshtavmware — APPROVED** [review]
LGTM

**Review by aaha97 — APPROVED** [review]
LGTM

**Pull request merged**